barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.cpp

```diff
@@ -456,12 +472,16 @@ AvmError Execution::execute_enqueued_call(AvmTraceBuilder& trace_builder,
     std::stack<uint32_t> debug_counter_stack;
     uint32_t counter = 0;
     trace_builder.set_call_ptr(context_id);
-    while ((pc = trace_builder.get_pc()) < bytecode.size()) {
+    while (is_ok(error) && (pc = trace_builder.get_pc()) < bytecode.size()) {
         auto [inst, parse_error] = Deserialization::parse(bytecode, pc);
 
-        // FIXME: properly handle case when an instruction fails parsing
-        // especially first instruction in bytecode
         if (!is_ok(error)) {
+            info("AVM failed to deserialize bytecode at pc: ", pc);
+            // FIXME: properly handle case when an instruction fails parsing!
+            // For now, we add a dummy row in main trace to mutate later.
+            // This error was encountered before any opcodes were executed, but
+            // we need at least one row in the execution trace to then mutate and say "it halted and consumed all gas!"
+            trace_builder.op_add(0, 0, 0, 0, OpCode::ADD_8);
             error = parse_error;
             break;
```

Comment:
Haven't tested this yet, but it should work! We can write a test for deserialization failures in a follow-up